srv/modules/runners/changed.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=8 expandtab shiftwidth=4 softtabstop=4
+
+import os.path
+import hashlib
+import logging
+
+"""
+This runner is here to detect config changes in the
+various configuration files to control service restarts.
+"""
+
+log = logging.getLogger(__name__)
+
+class Config(object):
+    def __init__(self, service_name):
+        self.base_dir = '/srv/salt/ceph/configuration/files/'
+        self.conf_dir = '/srv/salt/ceph/configuration/files/ceph.conf.d/'
+        self.checksum_dir = '/srv/salt/ceph/configuration/files/ceph.conf.checksum/'
+        self.checksum_file = self.checksum_dir + service_name + '.conf'
+        self.service_name = service_name
+        self.service_conf_files = [self.conf_dir + service_name + '.conf']
+        self.extra_files()
+        self.dependencies = self.dependencies()
+
+    def dependencies(self):
+        """
+        Services might depend on each other and have to trigger a restart.
+        I.e. Changes in the MDS section might affect the NFS-Ganesha service
+        hence the service should also be restarted.
+        # TODO for other services and complete tree
+        """
+        return { 'rgw': ['rgw', 'global_'],
+                 'mds': ['mds'],
+                 'mon': ['mon'],
+                 'osd': ['osd'],
+                 'client': ['client'],
+                 'global_': ['global_'],
+               }
+
+    def extra_files(self):
+        """
+        As we are not consistent in using the ceph.conf.d directory for
+        all the configs, we need this workaround until the initial issue
+        is fixed. REF: TODO: Deepsea issue reference
+        """
+        if self.service_name == 'rgw':
+            self.service_conf_files.append(self.base_dir + 'ceph.conf.rgw')
```

Comment:
should we just glob over all the conf.* files?

Reply:
That would also include the ceph.conf.rbd. I don't think that we want this to trigger a restart of RGW. Or do we?

But this is the question, what filechange triggers which service restart.

Reply:
since you're in a runner, we could use a strategy similar to ui_rgw as well, which is glob over only the files which are configured for rgw,

Reply:
I think I might be missing something. Do we have potentially more than the ceph.conf.d/rgw.conf, ceph.conf.rgw, ceph.conf.rgw-ssl ?

Reply:
so, we can use a custom rgw configuration, and the file name we use would be matching the value of ceph.conf.`rgw_configurations`, see the endpoints in `ui_rgw.py` and also the way we build configuration in /srv/salt/configuration/files/ceph.conf.j2, if we get not match then we default to `rgw`

Reply:
The ceph.conf.j2 will use ceph.conf.d for rgw files as well, but there filenames do not match the others.  I was going to suggest a separate PR to make the names consistent, but I am starting to think maybe we should do it here as well.  That is, ceph.rgw.conf becomes rgw.conf and we use self.conf_dir throughout.